### PR TITLE
[ISSUE #5157]Optimize AutoSwitchHAConnection HandShake's data protocol

### LINKED ISSUE #5157: Optimize AutoSwitchHAConnection send HandShake protocol
when Broker master received Broker master handshake data, then Broker master will send back the handshake data to broker slave.
image
EpochStartOffset and Additional not need in handshake. So need to remove  two fieds to reduce data transfer.and Keep it consistent with the documentation

2. Please tell us about your environment:
branch: develop

### PR DESCRIPTION
**Pull request by mxsm**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

close #5157 

## Brief changelog

- Optimize AutoSwitchHAConnection HandShake's data protocol

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAClient.java [hunk 1]
```diff
@@ -444,27 +444,32 @@ protected boolean processReadResult(ByteBuffer byteBufferRead) {
...
+                            AutoSwitchHAClient.this.processPosition += headerSize + bodySize;
                             AutoSwitchHAClient.this.waitForRunning(1);
                             LOGGER.error("State not matched, masterState:{}, slaveState:{}, bodySize:{}, offset:{}, masterEpoch:{}, masterEpochStartOffset:{}, confirmOffset:{}",
                                 masterState, AutoSwitchHAClient.this.currentState, bodySize, masterOffset, masterEpoch, masterEpochStartOffset, confirmOffset);
-                            return true;
+                            return false;
                         }
 
-                        if (diff >= (AutoSwitchHAConnection.MSG_HEADER_SIZE + bodySize)) {
+                        if (diff >= AutoSwitchHAConnection.TRANSFER_HEADER_SIZE + bodySize || diff >= AutoSwitchHAConnection.HANDSHAKE_HEADER_SIZE + bodySize) {
```
Comment (Oliverwqcwrw) [inline store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAClient.java:469]:
```suggestion
                        if (diff >= AutoSwitchHAConnection.HANDSHAKE_HEADER_SIZE + bodySize) {
```

Is it ok to keep only one condition?

**Review by Oliverwqcwrw — CHANGES_REQUESTED** [review]

File: store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAClient.java (hunk 1 above)
Comment (mxsm) [reply store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAClient.java:469]: Hi @Oliverwqcwrw  this Method handle handshake and transfer data. 
```
if (diff >= AutoSwitchHAConnection.TRANSFER_HEADER_SIZE + bodySize || diff >= AutoSwitchHAConnection.HANDSHAKE_HEADER_SIZE + bodySize)
```
So this condition requires judging both cases: handshake and transfer message. And `diff >= AutoSwitchHAConnection.TRANSFER_HEADER_SIZE + bodySize` must condition in front of `diff >= AutoSwitchHAConnection.HANDSHAKE_HEADER_SIZE + bodySize`

Comment (Oliverwqcwrw) [reply store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAClient.java:469]: get it, If this condition is not added, `currentState` may be `TRANSFER`, but diff is less than `TRANSFER_HEADER_SIZE + body size`

**Review by ni-ze — APPROVED** [review]
LGTM～

**Review by ShannonDing — APPROVED** [review]

**Review by tsunghanjacktsai — DISMISSED** [review]

**Review by Oliverwqcwrw — APPROVED** [review]

**Comment by mxsm** [comment]
@RongtongJin  PTAL~~

**Comment by hzh0425** [comment]
@LGTM  cc @RongtongJin

**Review by hzh0425 — APPROVED** [review]

**Review by hzh0425 — APPROVED** [review]
LGTM

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**